main.go

```diff
@@ -370,9 +344,35 @@ func doRun(filePath string) error {
 }
 
 // navLeft navigates to the next item to the left
+// After the navigation sequence (ArrowLeft) is sent, wait for a Location change.
+// If however, after 100ms (maxIterations*miniTick), no change is seen,
+//  then return (we have reached the end)
+// navLeft almost always exits after a single iteration,
+//  but without waiting for the location change, we often see
+//  a failure to navigate, especially on the first invocation,
+//  which causes a (false) early termination of the main navN loop
 func navLeft(ctx context.Context) error {
+	var prevLocation string
+	if err := chromedp.Location(&prevLocation).Do(ctx); err != nil {
+		return err
+	}
+
 	chromedp.KeyEvent(kb.ArrowLeft).Do(ctx)
-	chromedp.WaitReady("body", chromedp.ByQuery)
+
+	maxIterations := 10
+	miniTick := 10 * time.Millisecond
+	var location string
+	for i := 0; i < maxIterations; i++ {
+
+		if err := chromedp.Location(&location).Do(ctx); err != nil {
+			return err
+		}
+		if location != prevLocation {
+			log.Printf("navLeft break at it:%d", i)
+			break
+		}
+		time.Sleep(miniTick)
+	}
```

Comment:
let's worry about that in another PR please.

Reply:
Ok, I will remove that change.

Unfortunately, the original navLeft does not work for me, wich makes testing rather hard.

I think it's time we just close this PR.
I am pretty lost in the threaded comments.

I'd still love to help out somehow, but I am not sure what the best way to have a productive conversation is.

Reply:
Ok, I will remove that change.

Unfortunately, the original navLeft does not work for me, wich makes testing rather hard.

I think it's time we just close this PR.
I am pretty lost in the threaded comments.

I'd still love to help out somehow, but I am not sure what the best way to have a productive conversation is.

Reply:
> Ok, I reverted the change to `navLeft`.
> Unfortunately, the original `navLeft` does not work for me, which makes testing rather hard. _(it falsely terminates early too often)_.

Can you create another PR for that fix please? we don't have to wait for the current one to be over. It can all be done in parallel.

> I'm also a bit lost in the threaded comments. Perhaps I'm not using the review tooling correctly.

I suspect it's both. We probably got carried away too many times, and github can be weird at times (I'm still much more comfortable myself in gerrit tbh). I suggest reloading the page, taking a big breath and rereading it all from the start, noting as you go along what is actually still a conversation, and what is not. :-)

> If you feel this is dragging on, we can also just close the PR.

Let's not close it please. I am still hopeful that we can make one (variant) of the solutions you provided work, i.e. replacing the screenshots with another one that is less gross, but that also does not use arbitrary sleeps. But if you are bored with that problem, no worries, I'll just keep on iterating myself on that PR. And of course feel free to open other (independent) PRs in parallel.

Reply:
Cool. I did a separate PR for `-headless` flag.
I will look at navLeft next, and I can circle back to this navToLast/End again after.